external/storm-kafka-client/src/main/java/org/apache/storm/kafka/bolt/KafkaBolt.java

```diff
@@ -119,7 +117,12 @@ public void prepare(Map stormConf, TopologyContext context, OutputCollector coll
             }
         }
 
-        producer = mkProducer(boltSpecifiedProperties);
+        synchronized (KafkaBolt.class) {
+            if (producer == null) {
+                producer = mkProducer(boltSpecifiedProperties);
```

Comment:
since you are touching this code, can you please rename the method to **mkProducer** to **newProducer** or **createProducer**.  I don't think mk is conventional and/or meaningful.